addons/sale_timesheet/models/account.py

```diff
@@ -26,22 +26,30 @@ class AccountAnalyticLine(models.Model):
     commercial_partner_id = fields.Many2one('res.partner', compute="_compute_commercial_partner")
     timesheet_invoice_id = fields.Many2one('account.move', string="Invoice", readonly=True, copy=False, help="Invoice created from the timesheet", index='btree_not_null')
     so_line = fields.Many2one(compute="_compute_so_line", store=True, readonly=False,
+        domain="""[
+            ('qty_delivered_method', 'in', ['analytic', 'timesheet']),
+            ('order_id.partner_id', '=', commercial_partner_id),
```

Comment:
We could use the related field `order_partner_id`

```suggestion
            ('order_partner_id', '=', commercial_partner_id),
```